### PR TITLE
Suit sensors can now be adjusted through the strip menu

### PR DESCRIPTION
**Pull request by Cenrus**
You can now change someones suit sensor settings using a button on the strip menu. This doesn't work on suits without any sensors, and it is impossible to change uniforms with locked sensors, such as the orange prison suit. Doing this will give an obvious message to anyone nearby.

:cl:
tweak: Suit sensors can now be changed in the strip menu.
/:cl:

### CODE REVIEW

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
this shouldnt be instant

**Comment by Cenrus** [comment]
Why not? Suit sensors are obscenely powerful and have absolutely no counters outside of EMPs, which a number of antags don't have access to. It's next to impossible to get away with murder when there's an AI, Warden and Paramedic watching the crew monitor at all times, ready to call out the very moment and position that someone starts taking damage in maintenance.

**Comment by heepox** [comment]
suit sensors pretty much force me to instantly strip anyone i robust in maint or just EMP them.

**Comment by heepox** [comment]
aslong as this is faster than stripping it should be good

**Comment by PowerfulBacon** [comment]
I wasnt thinking about instantly turning them off on someone, more instantly turning them on. But both are pretty bad.

**Comment by PowerfulBacon** [comment]
Completely unrelated to this PR, but if you want to nerf suit sensors make them get affected by radio jammers or something

**Comment by ItsTisme** [comment]
> Completely unrelated to this PR, but if you want to nerf suit sensors make them get affected by radio jammers or something

much better idea imo, though a easy way to turn on suit sensors when a docter treated some one would be nice

**Comment by utilbird** [comment]
It is pretty annoying to have to almost fully strip someone to remove their crew sensor status. But there should definitely be an action timer for this, two seconds at least imo

**Comment by PowerfulBacon** [comment]
feel free to reopen once requested things are covered

**Comment by Crossedfall** [comment]
Re-opening on request

**Comment by Cenrus** [comment]
Added two second delay to sensor changes as requested

**Comment by bloons3** [comment]
Only 2 seconds seems a bit slow compared to having to get a hardsuit then jumpsuit off of someone to fully obscure their death.

**Comment by jupyterkat** [comment]
make it so spacesuits block the sensor button, or anything worn in the suit slot

**Comment by Cenrus** [comment]
> make it so spacesuits block the sensor button

It is obstructed by hardsuits and the like

**Comment by bloons3** [comment]
Can you add logging for changed suit sensors to the attacker's and defender's attack logs?

File: code/modules/clothing/clothing.dm [hunk 1]
```diff
@@ -195,48 +195,75 @@ BLIND     // can't see anything
 	female_clothing_icon 			= fcopy_rsc(female_clothing_icon)
 	GLOB.female_clothing_icons[index] = female_clothing_icon
 
-/obj/item/clothing/under/verb/toggle()
-	set name = "Adjust Suit Sensors"
-	set category = "Object"
-	set src in usr
-	var/mob/M = usr
-	if (istype(M, /mob/dead/))
+#define SENSOR_CHANGE_DELAY 1.5 SECONDS
```
Comment (bloons3) [inline code/modules/clothing/clothing.dm]: isn't the convention for defines to be at the top of the file?

File: code/modules/clothing/clothing.dm [hunk 2]
```diff
@@ -195,48 +195,75 @@ BLIND     // can't see anything
...
+	var/mob/M = user
+	if(isdead(M))
 		return
-	if (!can_use(M))
+	if(!can_use(M))
 		return
 	if(src.has_sensor == LOCKED_SENSORS)
-		to_chat(usr, "The controls are locked.")
+		to_chat(user, "<span class='warning'>The controls are locked.</span>")
 		return 0
```
Comment (bloons3) [inline code/modules/clothing/clothing.dm]: return TRUE/FALSE

File: code/modules/clothing/clothing.dm [hunk 3]
```diff
@@ -195,48 +195,75 @@ BLIND     // can't see anything
...
+		to_chat(user, "<span class='warning'>You have moved too far away!</span>")
 		return
-	sensor_mode = modes.Find(switchMode) - 1
+	var/sensor_selection = modes.Find(switchMode) - 1
 
-	if (src.loc == usr)
-		switch(sensor_mode)
+	if (src.loc == user)
+		switch(sensor_selection)
 			if(0)
```
Comment (bloons3) [inline code/modules/clothing/clothing.dm]: define for these magic numbers?

**Review by PowerfulBacon — DISMISSED** [review]

File: code/modules/clothing/clothing.dm [hunk 4]
```diff
@@ -195,48 +201,71 @@ BLIND     // can't see anything
...
 
-/obj/item/clothing/under/verb/toggle()
-	set name = "Adjust Suit Sensors"
-	set category = "Object"
-	set src in usr
-	var/mob/M = usr
-	if (istype(M, /mob/dead/))
+/obj/item/clothing/under/proc/set_sensors(mob/user)
+	var/mob/M = user
+	if(isdead(M))
```
Comment (PowerfulBacon) [inline code/modules/clothing/clothing.dm]: Check consciousness instead.

**Comment by Cenrus** [comment]
Checks consciousness stat now

File: code/modules/clothing/clothing.dm [hunk 5]
```diff
@@ -195,48 +201,71 @@ BLIND     // can't see anything
...
 
-/obj/item/clothing/under/verb/toggle()
-	set name = "Adjust Suit Sensors"
-	set category = "Object"
-	set src in usr
-	var/mob/M = usr
-	if (istype(M, /mob/dead/))
+/obj/item/clothing/under/proc/set_sensors(mob/user)
+	var/mob/M = user
+	if(M.stat != CONSCIOUS)
```
Comment (PowerfulBacon) [inline code/modules/clothing/clothing.dm]: use M.concious or M.isconcious its one of the 2

Comment (Cenrus) [reply code/modules/clothing/clothing.dm]: Turns out m.stat works fine

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**